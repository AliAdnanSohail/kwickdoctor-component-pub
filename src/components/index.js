export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
