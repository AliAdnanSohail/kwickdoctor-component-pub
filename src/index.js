export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
