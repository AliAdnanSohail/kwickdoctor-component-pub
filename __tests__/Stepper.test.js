import React, { Component, Fragment } from 'react';
import { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';

import { Button, Stepper, Step } from '../src';

const StepContent = props => (
  <Fragment>
    <Button
      onClick={() => {
        props.onChangeActiveStep();
      }}
    >
      Next
    </Button>
    <br />
    {`current step ${props.index + 1}`}
  </Fragment>
);

class StepperWrapper extends Component {
  constructor(props) {
    super(props);

    this.state = { activeStep: 0 };
  }

  handleChangeActiveStep = (index) => {
    this.setState({ activeStep: index });
  };

  render() {
    const { activeStep } = this.state;

    return (
      <Stepper activeStep={activeStep}>
        <Step>
          <StepContent onChangeActiveStep={() => this.handleChangeActiveStep(1)} />
        </Step>
        <Step>
          <StepContent onChangeActiveStep={() => this.handleChangeActiveStep(2)} />
        </Step>
        <Step>
          <StepContent onChangeActiveStep={() => this.handleChangeActiveStep(3)} />
        </Step>
        <Step>
          <StepContent onChangeActiveStep={() => this.handleChangeActiveStep(4)} />
        </Step>
        <Step>
          <StepContent onChangeActiveStep={() => this.handleChangeActiveStep(5)} />
        </Step>
        <Step>
          <StepContent onChangeActiveStep={() => this.handleChangeActiveStep(0)} />
        </Step>
      </Stepper>
    );
  }
}

it('renders correctly', () => {
  const element = mount(<StepperWrapper />);

  expect(EnzymeToJson(element)).toMatchSnapshot();
});

it('changing step works correctly', () => {
  const element = mount(<StepperWrapper />);

  expect(element.find('.step').text()).toEqual('Nextcurrent step 1');

  element.find('.button').simulate('click');

  const timeout = setTimeout(() => {
    expect(element
      .find('.step-icon')
      .first()
      .hasClass('.step-icon--completed')).toEqual(true);

    expect(element
      .find('.step-icon')
      .childAt(1)
      .hasClass('.step-icon--current')).toEqual(true);

    expect(element.find('.step').text()).toEqual('Nextcurrent step 2');
  }, 300);

  clearTimeout(timeout);
});
