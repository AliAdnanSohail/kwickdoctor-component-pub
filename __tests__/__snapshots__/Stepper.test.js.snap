// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<StepperWrapper>
  <Stepper
    activeStep={0}
  >
    <div>
      <StepperHeader
        active={0}
        total={3}
      >
        <ul
          className="jsx-291528123 stepper-header"
        >
          <li
            className="jsx-1340858167 step-icon step-icon--current"
            key="0"
          >
            <span
              className="jsx-1340858167"
            />
            <JSXStyle
              css={
                Array [
                  ".step-icon.jsx-1340858167{width:34px;height:34px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;z-index:1;}",
                  ".step-icon.jsx-1340858167 span.jsx-1340858167{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background:#eff0f4;border-radius:50%;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167{width:34px;height:34px;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167 svg{height:16px;width:16px;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167 svg polyline{stroke:#6fcb83 !important;stroke-width:5;}",
                  ".step-icon--current.jsx-1340858167 span.jsx-1340858167{width:24px;height:24px;}",
                  ".step-icon--current.jsx-1340858167 span.jsx-1340858167:before{content:'';height:10px;width:10px;background-color:#0c97f9;border-radius:50%;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
                  ".step-icon--next.jsx-1340858167 span.jsx-1340858167{width:12px;height:12px;}",
                ]
              }
              styleId="1340858167"
            />
          </li>
          <li
            className="jsx-1340858167 step-icon step-icon--next"
            key="1"
          >
            <span
              className="jsx-1340858167"
            />
            <JSXStyle
              css={
                Array [
                  ".step-icon.jsx-1340858167{width:34px;height:34px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;z-index:1;}",
                  ".step-icon.jsx-1340858167 span.jsx-1340858167{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background:#eff0f4;border-radius:50%;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167{width:34px;height:34px;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167 svg{height:16px;width:16px;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167 svg polyline{stroke:#6fcb83 !important;stroke-width:5;}",
                  ".step-icon--current.jsx-1340858167 span.jsx-1340858167{width:24px;height:24px;}",
                  ".step-icon--current.jsx-1340858167 span.jsx-1340858167:before{content:'';height:10px;width:10px;background-color:#0c97f9;border-radius:50%;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
                  ".step-icon--next.jsx-1340858167 span.jsx-1340858167{width:12px;height:12px;}",
                ]
              }
              styleId="1340858167"
            />
          </li>
          <li
            className="jsx-1340858167 step-icon step-icon--last step-icon--next"
            key="2"
          >
            <span
              className="jsx-1340858167"
            />
            <JSXStyle
              css={
                Array [
                  ".step-icon.jsx-1340858167{width:34px;height:34px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;z-index:1;}",
                  ".step-icon.jsx-1340858167 span.jsx-1340858167{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background:#eff0f4;border-radius:50%;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167{width:34px;height:34px;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167 svg{height:16px;width:16px;}",
                  ".step-icon--completed.jsx-1340858167 span.jsx-1340858167 svg polyline{stroke:#6fcb83 !important;stroke-width:5;}",
                  ".step-icon--current.jsx-1340858167 span.jsx-1340858167{width:24px;height:24px;}",
                  ".step-icon--current.jsx-1340858167 span.jsx-1340858167:before{content:'';height:10px;width:10px;background-color:#0c97f9;border-radius:50%;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
                  ".step-icon--next.jsx-1340858167 span.jsx-1340858167{width:12px;height:12px;}",
                ]
              }
              styleId="1340858167"
            />
          </li>
          <JSXStyle
            css={
              Array [
                ".stepper-header.jsx-291528123{position:relative;height:34px;margin:0;padding:16px 0;list-style:none;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}",
                ".stepper-header.jsx-291528123:before{content:'';position:absolute;left:11px;right:11px;top:32px;height:2px;background:#eff0f4;z-index:0;}",
              ]
            }
            styleId="291528123"
          />
        </ul>
      </StepperHeader>
      <Step
        active={true}
        index={0}
        key=".0"
      >
        <Transition
          appear={false}
          enter={true}
          exit={true}
          in={true}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={true}
        >
          <div
            className="jsx-4224162652 step"
            style={
              Object {
                "opacity": 1,
                "transform": "translateX(0)",
              }
            }
          >
            <StepOneContent
              onChangeActiveStep={[Function]}
            >
              <Button
                accent={false}
                className=""
                danger={false}
                disabled={false}
                flat={false}
                icon={null}
                loading={false}
                onClick={[Function]}
                rounded={false}
                squared={false}
                transparent={false}
              >
                <button
                  className="jsx-1748154264 button"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    className="button__content"
                  >
                    Next
                  </span>
                  <JSXStyle
                    css={
                      Array [
                        ".button.jsx-1748154264{position:relative;padding:13px 64px;height:52px;background-color:#0c97f9;box-sizing:border-box;cursor:pointer;color:#ffffff;line-height:26px;-webkit-transition:all 0.4s ease-in-out;transition:all 0.4s ease-in-out;overflow:hidden;border-width:1px;border-style:solid;border-color:#0c97f9;border-radius:26px;font-style:normal;font-weight:500;font-size:1rem;text-align:center;text-transform:uppercase;}",
                        ".button.jsx-1748154264:focus{outline:none;}",
                        ".button.jsx-1748154264 svg{fill:#ffffff;stroke:#ffffff;width:24px;height:24px;}",
                        ".button.jsx-1748154264 svg circle,path{stroke:#ffffff;}",
                        ".button__content{display:inline-block;-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;}",
                        ".button__wave.jsx-1748154264{position:absolute;display:block;width:0;height:0;border-radius:50%;background-color:#ffffff;opacity:0.5;-webkit-transform:translate(-50%,-50%);-ms-transform:translate(-50%,-50%);transform:translate(-50%,-50%);}",
                        ".button--loading.jsx-1748154264 .button__content{-webkit-transform:translateX(16px);-ms-transform:translateX(16px);transform:translateX(16px);}",
                        ".button--disabled.jsx-1748154264{opacity:0.7;cursor:not-allowed;}",
                        ".button.jsx-1748154264:not(.button--disabled):hover,.button.jsx-1748154264:not(.button--disabled):focus{background-color:#38a9fa;border-color:#38a9fa;box-shadow:2px 2px 10px rgba(0,0,0,0.1);}",
                        ".button--flat.jsx-1748154264{color:#0c97f9;background-color:transparent;border-color:transparent;}",
                        ".button--flat.jsx-1748154264 svg{stroke:#2A2A30;fill:#2A2A30;}",
                        ".button--flat.jsx-1748154264 svg path,circle{stroke:#2A2A30;}",
                        ".button--flat.jsx-1748154264:not(.button--disabled):hover,.button--flat.jsx-1748154264:not(.button--disabled):focus{color:#0c97f9;background-color:transparent;border-color:transparent;box-shadow:none;}",
                        ".button--accent.jsx-1748154264{color:#0c97f9;background-color:transparent;border-color:transparent;}",
                        ".button--accent.jsx-1748154264 svg{stroke:#2A2A30;fill:#2A2A30;}",
                        ".button--accent.jsx-1748154264 svg path,circle{stroke:#2A2A30;}",
                        ".button--accent.jsx-1748154264:not(.button--disabled):hover,.button--accent.jsx-1748154264:not(.button--disabled):focus{color:#0c97f9;background-color:transparent;border-color:transparent;}",
                        ".button--danger.jsx-1748154264{color:#ffffff;background-color:#ea6577;border-color:#ea6577;}",
                        ".button--danger.jsx-1748154264:not(.button--disabled):hover,.button--danger.jsx-1748154264:not(.button--disabled):focus{background-color:#ed7888;border-color:#ed7888;}",
                        ".button--rounded.jsx-1748154264{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;width:52px;height:52px;padding:0 !important;border-radius:50%;}",
                        ".button.jsx-1748154264:not(.button--rounded) .button__loading-icon{position:absolute;top:calc(50% - 12px);left:32px;}",
                        ".button--squared.jsx-1748154264{border-radius:12px;}",
                        ".button--transparent.jsx-1748154264{background-color:transparent;box-shadow:0px 0px 10px rgba(12,151,249,0.15);color:#2A2A30;}",
                        ".button--transparent.jsx-1748154264:not(.button--disabled):hover,.button--transparent.jsx-1748154264:not(.button--disabled):focus{color:#ffffff;}",
                        ".button--transparent.jsx-1748154264 svg{stroke:#2A2A30;fill:#2A2A30;}",
                        ".button--transparent.jsx-1748154264 svg path,circle{stroke:#2A2A30;}",
                        ".button--xs.jsx-1748154264{height:32px;line-height:1;padding:0 32px;font-size:0.75rem;}",
                        ".button--xs.button--rounded.jsx-1748154264{width:32px;height:32px;}",
                        ".button--xs.jsx-1748154264 svg{width:18px;height:18px;}",
                        ".button--xs.jsx-1748154264:not(.button--xs--rounded) .button__loading-icon{left:16px;top:calc(50% - 9px);}",
                        ".button--s.jsx-1748154264{height:42px;line-height:1;padding:0 42px;font-size:0.875rem;}",
                        ".button--s.button--rounded.jsx-1748154264{width:42px;height:42px;}",
                        ".button--s.jsx-1748154264 svg{width:18px;height:18px;}",
                        ".button--s.jsx-1748154264:not(.button--s--rounded) .button__loading-icon{left:21px;top:calc(50% - 9px);}",
                        ".button.has-clicked.jsx-1748154264 .button__wave.jsx-1748154264{-webkit-animation:wave-jsx-1748154264 0.8s;animation:wave-jsx-1748154264 0.8s;}",
                        "@-webkit-keyframes wave-jsx-1748154264{0%{opacity:0.5;width:0;height:0;}100%{opacity:0;width:300px;height:300px;}}",
                        "@keyframes wave-jsx-1748154264{0%{opacity:0.5;width:0;height:0;}100%{opacity:0;width:300px;height:300px;}}",
                      ]
                    }
                    styleId="1748154264"
                  />
                </button>
              </Button>
              <br />
              Step one
            </StepOneContent>
            <JSXStyle
              css={
                Array [
                  ".step.jsx-4224162652{-webkit-transition:all 0.3s ease-in-out;transition:all 0.3s ease-in-out;-webkit-transform:translateX(40px);-ms-transform:translateX(40px);transform:translateX(40px);opacity:0;}",
                ]
              }
              styleId="4224162652"
            />
          </div>
        </Transition>
      </Step>
      <Step
        active={false}
        index={1}
        key=".1"
      >
        <Transition
          appear={false}
          enter={true}
          exit={true}
          in={false}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={true}
        />
      </Step>
      <Step
        active={false}
        index={2}
        key=".2"
      >
        <Transition
          appear={false}
          enter={true}
          exit={true}
          in={false}
          mountOnEnter={false}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={true}
        />
      </Step>
    </div>
  </Stepper>
</StepperWrapper>
`;
